src/app/wallet/views/account_list.nim

```diff
@@ -99,3 +99,18 @@ QtObject:
   proc forceUpdate*(self: AccountList) =
     self.beginResetModel()
     self.endResetModel()
+
+  proc hasAccount*(self: AccountList, address: string): bool =
+    result = self.accounts.anyIt(it.account.address == address)
+
+  proc updateAssetsInList*(self: AccountList, address: string, assets: seq[Asset]) =
+    if not self.hasAccount(address):
+      return
+
+    let topLeft = self.createIndex(0, 0, nil)
+    let bottomRight = self.createIndex(self.accounts.len, 0, nil)
+    self.accounts.apply(proc(it: var AccountView) =
```

Comment:
You seem to call `updateAssetsInList` on every account in the list and then in `updateAssetsInList` you loop the accounts to find the right one.
Couldn't you just chnage the function to change  the assets on all the accounts straight away instead of only one? You'd get rid of one loop.

Reply:
Yes agreed, it likely would be a bit more efficient that way, but we'd effectively be updating all the account data, and have to refresh the UI accordingly (ie with `beginResetModel`). By doing it this way, we only really need to update the assets in the UI (via `self.dataChanged(topLeft, bottomRight, @[AccountRoles.Assets.int])`). I don't which one is more efficient: a loop in nim or refreshing more UI data 🤷‍♂️

Reply:
It's a good point, but right now you refresh all UI data anyway, because you do ` let bottomRight = self.createIndex(self.accounts.len, 0, nil)`, which targets the whole data set.
If you wanted to refresh only one row, you'd want to do something like this instead:
```nim
self.accounts.apply(proc(it: var AccountView) =
  if it.account.address == address:
    it.assets.setNewData(assets))
    let bottomRight = self.createIndex(it.index, 0, nil) # I'm not sure how to get the index
```
I'm not sure how to get the index, but you should get the drill

Reply:
True, all **rows** are targeted, but only the assets data in **each** row is being updated this way.